src/plugins/console/public/application/hooks/use_send_current_request_to_opensearch/use_send_current_request_to_opensearch.ts

```diff
@@ -112,5 +120,14 @@ export const useSendCurrentRequestToOpenSearch = (dataSourceId?: string) => {
         });
       }
     }
-  }, [dispatch, http, dataSourceId, settings, notifications.toasts, trackUiMetric, history]);
+  }, [
+    dispatch,
+    http,
+    dataSourceId,
+    settings,
+    notifications.toasts,
+    trackUiMetric,
+    history,
+    uiSettings,
```

Comment:
uiSettings is a service, would it ever change?

Reply:
Most of these are; not sure if it is because `useServicesContext` can throw (which would prevent code advancing here), linting rules, or something else but most of the services fetched from the context are passed in here too. Didn't want to re-engineer this section so followed the same pattern.